src/slices/sort.go

```diff
@@ -28,6 +28,18 @@ func SortFunc[S ~[]E, E any](x S, cmp func(a, b E) int) {
 	pdqsortCmpFunc(x, 0, n, bits.Len(uint(n)), cmp)
 }
 
+// SortFuncBasedOnBool sorts the slice x in ascending order as determined by the cmp
+// function. This sort is not guaranteed to be stable.
+// cmp(a, b) should return a boolean i.e a < b, true when
+// a > b and false when a <= b
+// SortFuncBasedOnBool requires that cmp is a strict weak ordering.
+// See https://en.wikipedia.org/wiki/Weak_ordering#Strict_weak_orderings.
+// Backported from https://pkg.go.dev/golang.org/x/exp/slices#SortFunc
+func SortFuncBasedOnBool[S ~[]E, E any](x S, cmp func(a, b E) bool) {
+	n := len(x)
+	pdqsortCmpFuncBasedonBool(x, 0, n, bits.Len(uint(n)), cmp)
+}
```

Comment:
uh what is this? I like the idea, but I believe the plan was just to simply remove the bool version totally. personally I prefer the bool version, but if its going into the standard library is probably needs a better name

Reply:
Please don’t reply on this GitHub thread. Visit [golang.org/cl/511395](https://go-review.googlesource.com/c/go/+/511395#message-38c5f335c9a6a4c1e10357c82a8855cdc41bffa2).

Reply:
nope, I am not doing that. I am signed in here, not there, so if you cant accept that then feel free to ignore my comments.